lib/rom/mapper.rb

```diff
@@ -56,7 +56,21 @@ def self.register_processor(processor)
     #
     # @api private
     def self.build(header = self.header, processor = :transproc)
-      new(Mapper.processors.fetch(processor).build(header), header)
+      if steps.empty?
+        new(Mapper.processors.fetch(processor).build(header), header)
+      else
+
+        step_transformers = steps.map do |s|
+          mapper = new(Mapper.processors.fetch(processor).build(s.header), s.header)
+          if mapper.transformer.is_a?(Transproc::Function) || mapper.transformer.is_a?(Transproc::Composite)
```

Comment:
Yeah good point, maybe I can make the assumption that transformers must have an interface for composition of `:+`.